src/components/Trade/hooks/useSwapper/utils.ts

```diff
@@ -0,0 +1,209 @@
+  const assetIdMap = Object.fromEntries(assetIds.map(assetId => [assetId, true]))
+  return assets.filter(asset => assetIdMap[asset.assetId])
+}
+
+export const getSendMaxAmount = (
+  sellAsset: Asset,
+  feeAsset: Asset,
+  quote: TradeQuote<KnownChainIds>,
+  sellAssetBalance: string,
+) => {
+  // Only subtract fee if sell asset is the fee asset
+  const isFeeAsset = feeAsset.assetId === sellAsset.assetId
+  const feeEstimate = bnOrZero(quote?.feeData?.fee)
+  // sell asset balance minus expected fee = maxTradeAmount
+  // only subtract if sell asset is fee asset
+  return fromBaseUnit(
+    bnOrZero(sellAssetBalance)
+      .minus(isFeeAsset ? feeEstimate : 0)
+      .toString(),
+    sellAsset.precision,
+  )
+}
+
+const getEvmFees = <T extends EvmChainId>(
+  trade: Trade<T> | TradeQuote<T>,
+  feeAsset: Asset,
+  tradeFeeSource: string,
+): DisplayFeeData<T> => {
+  const feeBN = bnOrZero(trade?.feeData?.fee).dividedBy(bn(10).exponentiatedBy(feeAsset.precision))
+  const fee = feeBN.toString()
+  const approvalFee = bnOrZero(trade.feeData.chainSpecific.approvalFee)
+    .dividedBy(bn(10).exponentiatedBy(feeAsset.precision))
+    .toString()
+  const totalFee = feeBN.plus(approvalFee).toString()
+  const gasPrice = bnOrZero(trade.feeData.chainSpecific.gasPrice).toString()
+  const estimatedGas = bnOrZero(trade.feeData.chainSpecific.estimatedGas).toString()
+
+  return {
+    fee,
+    chainSpecific: {
+      approvalFee,
+      gasPrice,
+      estimatedGas,
+      totalFee,
+    },
+    tradeFee: trade.feeData.tradeFee,
+    tradeFeeSource,
+  } as DisplayFeeData<T>
+}
+
+export const getFormFees = ({
+  trade,
+  sellAsset,
+  tradeFeeSource,
+  feeAsset,
+}: GetFormFeesArgs): DisplayFeeData<KnownChainIds> => {
+  const feeBN = bnOrZero(trade?.feeData?.fee).dividedBy(bn(10).exponentiatedBy(feeAsset.precision))
+  const fee = feeBN.toString()
+  const { chainNamespace } = fromAssetId(sellAsset.assetId)
+  switch (chainNamespace) {
+    case CHAIN_NAMESPACE.Ethereum:
+      return getEvmFees(
+        trade as Trade<EvmChainId> | TradeQuote<EvmChainId>,
+        feeAsset,
+        tradeFeeSource,
+      )
+    case CHAIN_NAMESPACE.Cosmos: {
+      return {
+        fee,
+        tradeFee: trade.feeData.tradeFee,
+        tradeFeeSource: trade.sources[0].name,
+      }
+    }
+    case CHAIN_NAMESPACE.Bitcoin: {
+      const utxoTrade = trade as Trade<UtxoSupportedChainIds>
+      return {
+        fee,
+        chainSpecific: utxoTrade.feeData.chainSpecific,
+        tradeFee: utxoTrade.feeData.tradeFee,
+        tradeFeeSource,
+      }
+    }
+    default:
+      throw new Error('Unsupported chain ' + sellAsset.chainId)
+  }
+}
+
+export const getBestSwapperFromArgs = async (
+  buyAssetId: AssetId,
+  sellAssetId: AssetId,
+  featureFlags: FeatureFlags,
+): Promise<Swapper<ChainId>> => {
+  const swapperManager = await getSwapperManager(featureFlags)
+  const swapper = await swapperManager.getBestSwapper({
```

Comment:
just be aware the current logic of "best swapper" is the 0th in the list

Reply:
Yes indeed. Not for this PR, but it'll get looked at 👌